feat: write client logs to tmp dir by default.

joshuef opened the pull request:
Also removes a swrm cmd log which would log full record values. (This is a libp2p behaviour, so best just not to log atm)

lionel1704 commented on src/messaging/serialisation/wire_msg_header.rs:
```diff
@@ -128,43 +132,63 @@ impl WireMsgHeader {
+            .map_err(|err| Error::Serialisation(format!("failed to serialize header: {}", err)))?;
 
-        Ok((buf_at_payload, header_size))
+        let serialized_size = u16::try_from(buffer_len - buffer.len()).unwrap();
```
Just flagging the `.unwrap()` here :eyes:

connec replied:
Yeah, I should tighten that up. Since the size limit was taken away, we can't rely on this holding (used to be that `serialized_size` *must* fit `u16`, now it's `usize`).

connec commented:
Some of this is now a bit egregious vs. #293, which achieves roughly the same thing far more simply and with no breakage. I'll hold on to the branch though, as I like the `MessageSkeleton` idea still.

joshuef committed:
feat: write client logs to tmp dir by default.

Also removes a swrm cmd log which would log full record values.
(This is a libp2p behaviour, so best just not to log atm)

grumbach commented on safenode/src/bin/safe/mod.rs:
```diff
@@ -21,7 +24,15 @@ async fn main() -> Result<()> {
+    let _log_appender_guard = init_node_logging(&Some(tmp_dir.join("safe-client.log")))?;
+
+    info!("Full client logs will be written to {:?}", tmp_dir);
+>>>>>>> Stashed changes
```
You've got some merge conflicts to resolve here

grumbach reviewed (approved):
LGTM
